tsdb/wlog/checkpoint.go

```diff
@@ -64,26 +64,50 @@ func LastCheckpoint(dir string) (string, int, error) {
 	return filepath.Join(dir, checkpoint.name), checkpoint.index, nil
 }
 
-// DeleteCheckpoints deletes all checkpoints in a directory below a given index.
+// CheckpointPrefix is the prefix used for checkpoint files.
+const CheckpointPrefix = "checkpoint."
+
+// CheckpointTempFileSuffix is the suffix used when creating temporary checkpoint files.
+const CheckpointTempFileSuffix = ".tmp"
+
+// DeleteCheckpoints deletes all checkpoints in a directory below a given index and any temporary checkpoints that might have
+// failed to be created.
 func DeleteCheckpoints(dir string, maxIndex int) error {
-	checkpoints, err := listCheckpoints(dir)
+	checkpoints, err := listCheckpoints(dir, true)
 	if err != nil {
 		return err
 	}
 
+	if len(checkpoints) == 0 {
+		return nil
+	}
+
+	// We can't guarantee that Checkpoint() will not be called at the same time as DeleteCheckpoints() so we only delete
+	// temporary checkpoints that are older than the most recent successful checkpoint.
```

Comment:
What is the scenario when there are multiple temporary checkpoints and we want to keep all of them?

Reply:
No scenario exists where we want to keep more than 1 temporary checkpoint. I'll reword this comment as it's rather awkward.

Reply:
I realized what you meant after I pushed. We shouldn't keep multiple temporary checkpoints even if they are past the latest.